scripts/Dockerfile.install-matrix

```
@@ -45,13 +57,21 @@ RUN apt-get update \
     && rm -rf /var/lib/apt/lists/*
 
 # Add the Raspberry Pi OS apt repo so Pi-only packages (liblgpio-dev,
-# chromium-headless-shell, ...) resolve. See the header note for why
-# [trusted=yes] is used.
-RUN echo "deb [trusted=yes] http://archive.raspberrypi.com/debian/ ${CODENAME} main" \
-    > /etc/apt/sources.list.d/raspi.list \
+# chromium-headless-shell, ...) resolve. Fetch the real archive GPG key via
+# curl and store it under /etc/apt/keyrings so `[signed-by=...]` works across
+# bullseye/bookworm/trixie without the `[trusted=yes]` escape hatch (which
+# older apt on bullseye doesn't fully honour — see header note above).
+RUN mkdir -p /etc/apt/keyrings \
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)